src/Symfony/Component/Ldap/Adapter/ExtLdap/PaginableQuery.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Ldap\Adapter\ExtLdap;
+
+use Symfony\Component\Config\Definition\Exception\Exception;
+
+/**
+ * Uses serverctrls to support pagination and removes
+ * the usage of the following deprecated methods:
+ * - ldap_control_paged_result: https://www.php.net/manual/en/function.ldap-control-paged-result.php
+ * - ldap_control_paged_result_response: https://www.php.net/manual/en/function.ldap-control-paged-result-response.php.
+ *
+ * Requires PHP 7.3 or greater
+ *
+ * @author Luca Saba <lucasaba@gmail.com>
+ */
+class PaginableQuery extends Query
```

Comment:
I'd prefer not adding a new class here, and merge the code in the existing Query class (undeprecating it): not using the deprecated functions should remain an "internal detail" concern, no need to make it leak to any public API (eg changing a class name)

Reply:
If all the logic of selecting the correct PHP version is placed in the Query class, the code looks... not very elegant.

I'll make a try.

Reply:
Leaking internal concerns is worse :)
`if (\PHP_VERSION_ID >= 70300) {` is fine, it will be optimized by opcache and will be easy to remove once the minimum supported version is bumped.